sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1240,6 +1263,16 @@ static void _configure_netdev(netdev_t *dev)
         DEBUG("gnrc_netif: enable NETOPT_TX_END_IRQ failed: %d\n", res);
     }
 #endif
+
+    uint16_t dev_type;
+    res = dev->driver->get(dev, NETOPT_DEVICE_TYPE, &dev_type, sizeof(dev_type));
+    assert(res == sizeof(dev_type));
+
+    switch(dev_type) {
+        case NETDEV_TYPE_IEEE802154:
+            _configure_netdev_802154(dev);
+            break;
```

Comment:
> Maybe there should be some ifdef for removing the dead code if we are building an application without 802.15.4 (e.g. Ethernet only), to reduce the rom usage

I can put a `#ifdef MODULE_NETDEV_IEEE802154` around the case here and the configuration function above.